scripts/anvil_tools/set_up_anvil_workspaces.py

```diff
@@ -0,0 +1,255 @@
+
+    return credentials.get_access_token().access_token
+
+
+def add_members_to_workspace(workspace_name, auth_domain_name, project="anvil_datastorage"):
+    """Add members to workspace permissions."""
+
+    acls = []
+    # add auth domain as READER, anvil-admins as OWNER
+    acls.append({'email': f'{auth_domain_name}@firecloud.org', 'accessLevel': 'READER', 'canShare': False, 'canCompute': False})
+    acls.append({'email': 'anvil-admins@firecloud.org', 'accessLevel': 'OWNER', 'canShare': True, 'canCompute': True})
+
+    json_request = json.dumps(acls)
+
+    # request URL for updateWorkspaceACL
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}/acl?inviteUsersNotFound=false"
+
+    # Get access token and and add to headers for requests.
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "*/*", "Content-Type": "application/json"}
+    # -H  "accept: */*" -H  "Authorization: Bearer [token] -H "Content-Type: application/json"
+
+    # capture response from API and parse out status code
+    response = requests.patch(uri, headers=headers, data=json_request)
+    status_code = response.status_code
+
+    emails = [acl['email'] for acl in acls]
+    # print success or fail message based on status code
+    if status_code != 200:
+        print(f"WARNING: Failed to update {project}/{workspace_name} with the following user(s)/group(s): {emails}.")
+        print("Check output file for error details.")
+        return status_code, response.json()
+    else:
+        print(f"Successfully updated {project}/{workspace_name} with the following user(s)/group(s): {emails}.")
+        return status_code, emails
+
+
+def check_workspace_exists(workspace_name, project="anvil-datastorage"):
+    """Determine if a workspace of given namespace/name already exists."""
+
+    # don't need full response - could be very large and time consuming
+    uri = f"https://api.firecloud.org/api/workspaces/{project}/{workspace_name}?fields=owners,workspace.createdBy,workspace.authorizationDomain"
+
+    # Get access token and and add to headers for requests.
+    # -H  "accept: application/json" -H  "Authorization: Bearer [token]
+    headers = {"Authorization": "Bearer " + get_access_token(), "accept": "application/json"}
+
+    # capture response from API and parse out status code
+    response = requests.get(uri, headers=headers)
+    status_code = response.status_code
+
+    if status_code == 404:  # if workspace does not exist
+        return False
+    elif status_code == 200:  # if workspace already exists
+        print(f"Workspace already exists with name: {project}/{workspace_name}.")
+        print(f"Existing workspace details: {json.dumps(response.json(), indent=2)}")
+        # make user decide if they want to update/overwrite existing workspace
+        update_existing_ws = input("Would you like to continue modifying the existing workspace? (Y/N)" + "\n")
+        return update_existing_ws.upper()
```

Comment:
if the user says 'Y', won't `if not status_ws_exists` still evaluate to False and the code will not continue past line 90? if you want this to evaluate to a boolean, you could instead here return `update_existing_ws.upper() == 'N'` and that will return False if they do want to continue, or True if they don't. double check that logic but I think you need something like that?